### PR TITLE
DatePicker: fixes issue #14523

### LINKED ISSUE #14523: [Bug Report] Date Picker When type = week and first DayOfWeek = 3, 2019-01-01 shows Wednesday, not Tuesday.
### Element UI version
2.4.11

### OS/Browsers version
chrome

### Vue version
2.6.7

### Reproduction Link
[https://jsfiddle.net/wLnje7t1/](https://jsfiddle.net/wLnje7t1/)

### Steps to reproduce
设置 DatePicker，
type = "week"，
picker-options="{firstDayOfWeek: 3}"，
查看日期为2019-01-01时的日期面板

### What is Expected?
2019-01-01显示为周二

### What is actually happening?
2019-01-01显示为周三

### PR DESCRIPTION
**Pull request by sinchang**
Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

Fixes #14523

### CODE REVIEW

File: packages/date-picker/src/basic/date-table.vue [hunk 1]
```diff
@@ -167,26 +167,28 @@
...
             const isToday = time === now;
+            const getFirstDayPosition = () => day + offset < 0 ? 7 + day + offset : i * 7 + j;
 
             if (isToday) {
               cell.type = 'today';
             }
 
             if (i >= 0 && i <= 1) {
-              if (j + i * 7 >= (day + offset)) {
+              const flag = day + offset < 0 ? 7 + day + offset : day + offset;
```
Comment (wacky6) [inline packages/date-picker/src/basic/date-table.vue]: what does `flag` mean. It should be meaningful.

Comment (sinchang) [reply packages/date-picker/src/basic/date-table.vue]: any suggestions?

File: packages/date-picker/src/basic/date-table.vue [hunk 2]
```diff
@@ -167,26 +167,28 @@
...
+              if (j + i * 7 >= flag) {
                 cell.text = count++;
                 if (count === 2) {
-                  firstDayPosition = i * 7 + j;
+                  firstDayPosition = getFirstDayPosition();
```
Comment (wacky6) [inline packages/date-picker/src/basic/date-table.vue]: It seems `firstDayPosition` is not used anywhere else.
I think you can remove it. (including `rows.firstDayPosition = firstDayPosition;` at line 218)

File: packages/date-picker/src/basic/date-table.vue (hunk 1 above)
Comment (wacky6) [reply packages/date-picker/src/basic/date-table.vue]: numberOfDaysFromPreviousMonth ?

Comment (sinchang) [reply packages/date-picker/src/basic/date-table.vue]: I have updated to `prevMonthElementsLength `, if not ok, I accept your idea.

Comment (wacky6) [reply packages/date-picker/src/basic/date-table.vue]: `number` seems better, it replaces elements & length.

IMO, the word `element` usually refers to DOM element, which is not rendered here.

Comment (sinchang) [reply packages/date-picker/src/basic/date-table.vue]: resolved. thanks.

**Pull request merged**